File: backend/app/schemas/config_sections/remove_ads_config.py
from pydantic import BaseModel, Field


class RemoveAdsConfig(BaseModel):
    """Remove Ads offer configuration"""
    enabled: bool = Field(default=True, description="Whether the remove ads offer is enabled")
    minLevel: int = Field(..., ge=1, description="Minimum level to show the offer")
    adWatchedTrigger: int = Field(..., ge=0, description="Number of ads watched to trigger the offer")
    daysPlayedTrigger: int = Field(..., ge=0, description="Days played to trigger the offer")
    durationHours: int = Field(..., ge=1, description="Duration of the offer in hours")
    maxLifetimeShows: int = Field(..., ge=1, description="Maximum lifetime shows")
    maxSessionShows: int = Field(..., ge=1, description="Maximum shows per session")
    cooldownPopupHours: int = Field(..., ge=0, description="Cooldown between popup shows in hours")
    cooldownOfferHours: int = Field(..., ge=0, description="Cooldown between offer shows in hours")
    

File: frontend/src/lib/validations/removeAdsConfig.ts
import { z } from 'zod';

export const removeAdsConfigSchema = z.object({
  enabled: z.boolean().default(true),
  minLevel: z.number({ required_error: 'Min Level is required' }).int().min(1),
  adWatchedTrigger: z.number({ required_error: 'Ad Watched Trigger is required' }).int().min(0),
  daysPlayedTrigger: z.number({ required_error: 'Days Played Trigger is required' }).int().min(0),
  durationHours: z.number({ required_error: 'Duration Hours is required' }).int().min(1),
  maxLifetimeShows: z.number({ required_error: 'Max Lifetime Shows is required' }).int().min(1),
  maxSessionShows: z.number({ required_error: 'Max Session Shows is required' }).int().min(1),
  cooldownPopupHours: z.number({ required_error: 'Cooldown Popup Hours is required' }).int().min(0),
  cooldownOfferHours: z.number({ required_error: 'Cooldown Offer Hours is required' }).int().min(0),
});

export type RemoveAdsConfig = z.infer<typeof removeAdsConfigSchema>;

export const defaultRemoveAdsConfig: RemoveAdsConfig = {
  enabled: false,
  minLevel: 0,
  adWatchedTrigger: 0,
  daysPlayedTrigger: 0,
  durationHours: 0,
  maxLifetimeShows: 0,
  maxSessionShows: 0,
  cooldownPopupHours: 0,
  cooldownOfferHours: 0,
};
